server/test/actions/ApplicantAuthorizationActionTest.java

```diff
@@ -0,0 +1,164 @@
+package actions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import auth.CiviFormProfile;
+import auth.ProfileUtils;
+import controllers.WithMockedProfiles;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import models.ApplicantModel;
+import models.ProgramModel;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import play.api.routing.HandlerDef;
+import play.mvc.Action;
+import play.mvc.Http;
+import play.mvc.Http.Request;
+import play.mvc.Result;
+import play.routing.Router;
+import play.test.Helpers;
+import repository.VersionRepository;
+import scala.jdk.javaapi.CollectionConverters;
+import support.ProgramBuilder;
+
+public class ApplicantAuthorizationActionTest extends WithMockedProfiles {
```

Comment:
it might also be good to test the program auth passes but user auth doesn't scenario as well.